avocado/core/test.py

```diff
@@ -133,7 +133,136 @@ def str_filesystem(self):
                                  % (self.str_uid, str(self)))
 
 
-class Test(unittest.TestCase):
+class BaseTestData(object):
+
+    """
+    Class that adds the hability for tests to have access to data files
+
+    An implementation of this interfaces needs to declare the name of
+    data sources it knows, and a single method for users to get the
+    data files, either from a specific source, or from its built-in
+    order of sources.
+    """
+
+    DATA_SOURCES = []
+
+    def _check_valid_data_source(self, source):
+        """
+        Utility to check if user chose a specific data source
+
+        :param source: either None for no specific selection or a source name
+        :type source: None or str
+        :raises: ValueError
+        """
+        if source is not None and source not in self.DATA_SOURCES:
+            msg = 'Data file source requested (%s) is not one of: %s'
+            msg %= (source, ', '.join(self.DATA_SOURCES))
+            raise ValueError(msg)
+
+    def get_data(self, filename, source=None, must_exist=True):
+        """
+        Retrieves the path to a given data file.
+
+        :param filename: the name of the data file to be retrieved
+        :type filename: str
+        :param source: one of the defined data sources
+        :type source: str
+        :param must_exist: whether the existence of a file is checked
+        :type must_exist: bool
+        :rtype: str or None
+        """
+        raise NotImplementedError
+
+
+class TestData(BaseTestData):
+
+    DATA_SOURCES = ["file", "test", "variant"]
```

Comment:
Wouldn't it be better to organize DATA_SOURCES in the order they are usually obtained in? It confused me first before I found you `reverse` it on every `get_data` call.

Reply:
Yep, actually I think so too.